bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/internal/DiscoveryResultImpl.java

```diff
@@ -48,44 +49,20 @@
     /**
      * Creates a new instance of this class with the specified parameters.
      *
-     * @param thingUID
-     *            the Thing UID to be set (must not be null). If a {@code Thing} disappears and is discovered again, the
-     *            same {@code Thing} ID
-     *            must be created. A typical {@code Thing} ID could be the
+     * @param thingTypeUID the {@link ThingTypeUID}
+     * @param thingUID the {@link ThingUID} to be set (must not be null). If a {@code Thing} disappears and is
+     *            discovered again, the same {@code Thing} ID must be created. A typical {@code Thing} ID could be the
      *            serial number. It's usually <i>not</i> a product name.
-     * @param properties the properties to be set (could be null or empty)
-     * @param representationProperty the representationProperty to be set (could be null or empty)
-     * @param label the human readable label to set (could be null or empty)
      * @param bridgeUID the unique bridge ID to be set
-     * @param timeToLive time to live in seconds
-     *
-     * @throws IllegalArgumentException
-     *             if the Thing type UID or the Thing UID is null
-     * @deprecated use {@link #DiscoveryResultImpl(ThingUID, ThingTypeUID, ThingUID, Map, String, String, long)}
-     *             instead.
-     */
-    @Deprecated
-    public DiscoveryResultImpl(ThingUID thingUID, ThingUID bridgeUID, Map<String, Object> properties,
-            String representationProperty, String label, long timeToLive) throws IllegalArgumentException {
-        this(thingUID.getThingTypeUID(), thingUID, bridgeUID, properties, representationProperty, label, timeToLive);
-    }
-
-    /**
-     * Creates a new instance of this class with the specified parameters.
-     *
-     * @param thingTypeUID the {@link ThingTypeUID}
-     * @param thingUID the Thing UID to be set (must not be null). If a {@code Thing} disappears and is discovered
-     *            again, the same {@code Thing} ID must be created. A typical {@code Thing} ID could be the serial
-     *            number. It's usually <i>not</i> a product name.
      * @param properties the properties to be set (could be null or empty)
      * @param representationProperty the representationProperty to be set (could be null or empty)
      * @param label the human readable label to set (could be null or empty)
-     * @param bridgeUID the unique bridge ID to be set
      * @param timeToLive time to live in seconds
      *
-     * @throws IllegalArgumentException
-     *             if the Thing type UID or the Thing UID is null
+     * @throws IllegalArgumentException if the Thing type UID or the Thing UID is null
```

Comment:
This description is confusing. The ctor does not throw an IAE if `ThingTypeUID` is `null` but if TTL is missing.

Reply:
Can you update it so it documents the actual implementation?

Reply:
Done.